vivisect/analysis/generic/pointers.py

```diff
@@ -62,30 +64,45 @@ def analyze(vw):
         try:
             logger.debug('pointer(4): 0x%x -> 0x%x', addr, pval)
             vw.makePointer(addr, follow=True)
-            done.append((addr, pval))
+            done[addr] = pval
         except Exception as e:
             logger.error('makePointer() failed for 0x%.8x (pval: 0x%.8x) (err: %s)', addr, pval, e)
 
-    # Now let's see what these guys should be named (if anything)
-    for _, _ in done:
```

Comment:
i really like your graph.
i'm wondering if you wouldn't achieve the same results and resultant speed-up by just removing the outer loop.

Reply:
So I tired that initially and a ton of things in testelf broke. This pointer graph stuff just kinda worked right out of the box.

Reply:
interesting.  i wonder what broke and why.

Reply:
Probably because if we've got A, B, and C, with C being some named things, and A pointing to B and B pointing to C, if we encounter the value pair (A, B) first in the loop, we'll skip it since it's got no name for us to go off of, and then we encounter the pair (B, C), and we make a name for B pointing to C, but without the outer `for _, _ in done` loop, we'd never backtrack to make a name for A.

The outer loop was comprehensive so that we would eventually populate all the possible pairs and names, but it was still N^2, so it was slow as hell.